usr/src/lib/libzpool/common/kernel.c

```diff
@@ -461,6 +461,16 @@ kernel_fini(void)
 	system_taskq_fini();
 }
 
+/* ARGSUSED */
+uint32_t
+zone_get_hostid(void *zonep)
```

Comment:
I'll need to look into it. It seems like `zone_get_hostid()` has a behaviour specific to libzpool as the value returned will either be `0` or `gethostid()` depending on the mode passed in `kernel_ini()` (see initialization of `hw_serial`). This might not be what we want for all users of libfakekernel.

Reply:
After further investigation I'd propose to keep this in `libzpool` as the generic `libfakekernel` implementation should probably just be calling `gethostid()`. I'll add a comment in `zone_get_hostid()` to explain this special behaviour.